apiserver/nodebootstrapconfig.go

```diff
@@ -0,0 +1,52 @@
+package apiserver
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"net/http"
+
+	agent "github.com/Azure/agentbaker/pkg/agent"
+	"github.com/Azure/agentbaker/pkg/agent/datamodel"
+)
+
+const (
+	// RoutePathNodeBootstrapping the route path to get node bootstrapping data.
+	RoutePathNodeBootstrapConfig string = "/nodebootstrapconfig"
+)
+
+func handleError(err error, w http.ResponseWriter) {
+	log.Println(err.Error())
+	http.Error(w, err.Error(), http.StatusBadRequest)
+}
+
+// GetNodeBootstrapConfig endpoint for getting node bootstrapping data.
+func (api *APIServer) GetNodeBootstrappingConfig(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
```

Comment:
do you know if there is a default context timeout here? I kind of doubt it, can you add one? can be longish, even 30s would be fine.